pages/_app.tsx

```diff
@@ -15,10 +15,12 @@ export default function App({ Component, pageProps }: AppProps) {
     if ('errors' in data) {
       console.log(data.errors);
       setUser(undefined);
+      sessionStorage.clear();
       return;
     }
 
     setUser(data.user);
+    sessionStorage.setItem('uid', data.user.id.toString());
```

Comment:
Since we discussed the reason behind this offline, copying here:

> This is to demonstrate that with a client-side only fix, the data is still passed in the http response, even though it's hidden on the page

This is great! 🙌

Maybe we can do the same thing I suggested for example 4, where we make the existing file into `vulnerable-1.tsx` and your new proposed client-side vulnerability into `vulnerable-2.tsx`

Then we can still show the glaring, obvious vulnerability first (in `vulnerable-1.tsx`) and then show how solving it only on the client side doesn't really solve the problem completely

Reply:
agree. will update it accordingly